woocommerce/woocommerce
# Add step buttons for Number block and component

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:






Closes #40803 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add the following as a mu-plugin:
```php
<?php

function YOUR_PREFIX_add_block( $product_name_block ) {
  $parent = $product_name_block->get_parent();

  $parent->add_block(
    array(
      'id' => 'test_id3',
      'blockName' => 'woocommerce/product-number-field',
      'attributes' => array(
        'label' => __( 'Field', 'woocommerce' ),
        'property' => 'meta_data.test3',
        'step' => 2,
        )
    )
  );
}

add_action( 'woocommerce_block_template_area_product-form_after_add_block_product-name', 'YOUR_PREFIX_add_block' );
```
2. Enable the new product editor at /wp-admin/admin.php?page=wc-settings&tab=advanced&section=features
3. Go to Products > Add
4. Check the newly added field on the Basic details section
5. Check the "up" and "down" buttons and click them.
6. Press "Up" and "Down" keyboard keys and check if it also works.




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
